predict.py

```diff
@@ -162,28 +160,24 @@ def predict(
 
         if replicate_weights:
             start = time.time()
```

Comment:
I think we actually don't want to `delete_lora` here. Our hot-swapping code (which loads loras into a base image at prediction time for fine-tunes) is set up s.t. multiple requests to the same model should be routed to the same base image - meaning that if we have this setup, we'll often delete a lora, then reload the same lora over and over. 

my preference here would be to leave this implementation to the `engine.set_lora` call on line 91, so that each engine can manage its loras as needed (the overhead of `delete_lora` may be unnecessary depending). Willing to argue about this - but even if we do `delete_lora`, should move it to after the check [here](https://github.com/replicate/cog-llama-template/blob/e8aedde841b5f12a83b79f882972d5be00367a90/predict.py#L89)

Reply:
Yeah, we've moved that from the engines implementation to the engine itself. Resolving!